docs/tpm2-pkcs11_object_auth_model.md

```diff
@@ -0,0 +1,528 @@
+encrypt the auth value of existing TPM2 objects to be assumed into the
+tpm2-pkcs11 model. This is a transient secondary object.
+4. USER(keys) object: The authorization model is same as the
+Wrapping-Key-Sealing object and thus requires the USER-PIN object auth value via
+a policy. These are user created keys and objects that are used in user
+operations like encryption, decryption, signing, verification, etc. These are
+secondary transient objects.
+
+## NV object diagram
+Overview of the use of the TPM2 NV indexes in the implementation.
+<p>
+  <img src="illustrations/object_diagram_nv_objects.png"
+</p>
+<br>
+<br>
+The illustration above is the key/object diagram showing the relationship of the
+tpm2-pkcs11 objects. The SO-PIN and USER-PIN objects are NV index objects while
+the Wrapping-Key-Sealing object and the USER(keys) objects are secondary
+transient objects.
+
+## NV policy diagram
+Overview of policies used to satisfy standards requirements with NV objects used
+as SO/ USER PIN reference objects.
+<p>
+  <img src="illustrations/policy_model_nv_objects.png"
+</p>
+<br>
+<br>
+The illustration above is the policy diagram showing the relationship of the
+authorization model used by the tpm2-pkcs11 objects where the SO-PIN and the
+USER-PIN objects are NV indexes and the Wrapping-Key-Sealing object and the USER
+(keys) objects are transient secondary objects.
+
+## Transient object diagram
+Overview of the use of TPM2 transient objects in the implementation.
+<p>
+  <img src="illustrations/object_diagram_transient_objects.png"
+</p>
+<br>
+<br>
+The illustration above is the key/object diagram showing the relationship of the
+tpm2-pkcs11 objects. The SO-PIN and USER-PIN objects along with the
+Wrapping-Key-Sealing object and the USER(keys) objects are secondary transient
+objects.
+
+## Transient policy diagram
+Overview of the policies used to satisfy standards requirements with TPM2
+transient object used as SO/ USER PIN reference objects.
+<p>
+  <img src="illustrations/policy_model_transient_objects.png"
+</p>
+<br>
+<br>
+The illustration above is the policy diagram showing the relationship of the
+authorization model used by the tpm2-pkcs11 objects where the SO-PIN and the
+USER-PIN objects along with the Wrapping-Key-Sealing object and the USER (keys)
+objects are transient secondary objects.
+
+## Important note on authorization
+While both authorization models discussed resolve the issue of isolation of SO
+and USER roles there is a serious problem with the authorization model where the
+SO-PIN and USER-PIN reference objects are transient secondary objects. In that,
+when the authorization of the objects is changed it creates a new sensitive
+portion of the object while the old sensitive information cannot be invalidated.
+To that effect, anyone who has a cached copy of an older object whose pin may
+have been compromised can still access the USER objects.
```

Comment:
This also violates the "While SO could change USER PIN, the SO cannot restore the old USER PIN." design consideration set out under "PKCS11 Roles". A malicious SO could create a second USER-PIN object for their own use, which the real USER would never know about.

Reply:
True. I will append this to the cautionary language.